feature-libs/customer-ticketing/root/events/customer-ticketing-event.listener.spec.ts

```diff
@@ -0,0 +1,79 @@
+import { TestBed } from '@angular/core/testing';
+import {
+  CurrencySetEvent,
+  CxEvent,
+  EventService,
+  LanguageSetEvent,
+  LoginEvent,
+  LogoutEvent,
+} from '@spartacus/core';
+import { Subject } from 'rxjs';
+import { CustomerTicketingEventListener } from './customer-ticketing-event.listener';
+import {
+  GetTicketQueryReloadEvent,
+  GetTicketQueryResetEvent,
+} from './customer-ticketing.events';
+import createSpy = jasmine.createSpy;
+
+const mockEventStream$ = new Subject<CxEvent>();
+
+class MockEventService implements Partial<EventService> {
+  get = createSpy().and.returnValue(mockEventStream$.asObservable());
+  dispatch = createSpy();
+}
+
+describe('CustomerTicketingEventListener', () => {
+  let eventService: EventService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        CustomerTicketingEventListener,
+        { provide: EventService, useClass: MockEventService },
+      ],
+    });
+
+    TestBed.inject(CustomerTicketingEventListener);
+    eventService = TestBed.inject(EventService);
+  });
+
+  describe('onGetTicketQueryReload', () => {
+    it('LanguageSetEvent should dispatch GetTicketQueryReloadEvent', () => {
+      mockEventStream$.next(new LanguageSetEvent());
+
+      expect(eventService.dispatch).toHaveBeenCalledWith(
+        {},
+        GetTicketQueryReloadEvent
+      );
```

Comment:
nit:
These lines are repeated: can we factorize them by:
function assertServiceDispatchForEvent(event, dispatchMethod) {
     mockEventStream$.next(event);

      expect(eventService.dispatch).toHaveBeenCalledWith(
        {},
        dispatchMethod
      );
}

Reply:
yes. changing